packages/hub/tasks/send-email-card-drop-verification.ts

```diff
@@ -1,7 +1,67 @@
 import { Helpers } from 'graphile-worker';
+import config from 'config';
+import { query } from '@cardstack/hub/queries';
+import { inject } from '@cardstack/di';
+import * as Sentry from '@sentry/node';
 
 export default class SendEmailCardDropVerification {
-  async perform(_payload: any, helpers: Helpers) {
-    helpers.logger.info(`Placeholder for SendEmailCardDropVerification`);
+  email = inject('email');
+  emailCardDropRequests = query('email-card-drop-requests', { as: 'emailCardDropRequests' });
+
+  async perform(
+    payload: {
+      id: string;
+      email: string;
+    },
+    helpers: Helpers
+  ) {
+    const request = (
+      await this.emailCardDropRequests.query({
+        id: payload.id,
+      })
+    )[0];
+
+    if (!request) {
+      let e = new Error(`Unable to find card drop request with id ${payload.id}`);
+      Sentry.captureException(e, {
+        tags: {
+          event: 'send-email-card-drop-verification',
+        },
+      });
+      return;
+    }
+
+    const params = new URLSearchParams();
+    params.append('eoa', request.ownerAddress);
+    params.append('verification-code', request.verificationCode);
+    params.append('email-hash', request.emailHash);
+    const verificationLink = config.get('cardDrop.verificationUrl') + '?' + params.toString();
+
+    const senderEmailAddress = config.get('aws.ses.supportEmail') as string;
+
+    const emailTitle = 'Claim your Card Drop';
+    const emailBodyHtml = `<h1></h1> This is your verification link: <a href="${verificationLink}">${verificationLink}</a>`;
+    const emailBodyText = `This is your verification link: ${verificationLink}`;
```

Comment:
Seeing this potential site of XSS (is it? or injection of some kind), is there any possibility that someone can submit a fake EOA?

Reply:
It does seem possible and we should probably sanitize this. Unclear whether SES does this for us, so will create a linear issue for this.